api/helpers/errors.js

```diff
@@ -1,208 +1,297 @@
-/* eslint-disable */
+/* eslint-disable max-classes-per-file */
+class ProblemDetail extends Error {
+  constructor(status, type, title, detail) {
+    super();
+    this.status = status;
+    this.type = type;
+    this.title = title;
+    this.message = detail;
+  }
+}
+
 // Thrown when parameter(s) are missing/invalid
 // See https://httpstatuses.com/422
-class InvalidEnvironmentConfiguration extends Error {
-  constructor(message) {
+class InvalidEnvironmentConfiguration extends ProblemDetail {
+  constructor(detail) {
     super();
-    this.name = "InvalidEnvironmentConfiguration";
-    this.message = message;
+    this.status = 422;
+    this.type = "invalid-environment-configuration";
+    this.title = "An environment variable is missing.";
+    this.message = detail;
   }
 }
 
-class UnableToCreatePatronWithAxios extends Error {
-  constructor(message) {
+class KMSDecryption extends ProblemDetail {
+  constructor(detail) {
     super();
-    this.name = "UnableToCreatePatronWithAxios";
-    this.message = message;
+    this.status = 500;
+    this.type = "aws-kms-decryption-error";
+    this.title = "AWS KMS decryption error";
+    this.message = detail;
   }
 }
 
-class InvalidRequest extends Error {
-  constructor(message) {
+class UnableToCreatePatronWithAxios extends ProblemDetail {
+  constructor(detail) {
     super();
-    this.name = "InvalidRequest";
+    this.status = 400;
+    this.type = "unable-to-create-patron";
+    this.title = "Unable to create patron with axios";
+    this.message = detail;
+  }
+}
+
+class InvalidRequest extends ProblemDetail {
+  constructor(detail) {
+    super();
+    this.status = 400;
     this.type = "invalid-request";
-    this.state = 400;
-    this.message = message;
+    this.title = "Invalid Request";
+    this.message = detail;
+    // To support older versions of API where client expect these values:
```

Comment:
All error objects will render `title` and `message` so it can be a standard javascript `Error` object. The `message` gets converted into a `detail` when it's returned to the client. In some cases, we _still_ want to return `name` and `message` to the client and that's what happens in the following lines. 

This gets repeated in other error classes.